Content.Shared/Damage/Systems/DamageableSystem.cs

```diff
@@ -362,6 +363,38 @@ private void DamageableHandleState(EntityUid uid, DamageableComponent component,
                 DamageChanged(uid, component, delta);
             }
         }
+
+        /// <summary>
+        /// Returns a string list containing the type/s of the highest damage value
+        /// </summary>
+        public  List<string> GetHighestDamageTypes(Dictionary<string, FixedPoint2> damagePerGroup, DamageSpecifier damage)
+        {
+            var highestType = new List<string>();
+            var highestValue = FixedPoint2.Zero;
+
+            foreach (var (damageGroupId, damageAmount) in damagePerGroup)  //go through each group
```

Comment:
It looks like this method can be simplified by a lot. It only needs the damage specifier which has all the info it needs (in its `.DamageDict`), and using that you can use some LINQ magic to do all this in one go:
`damage.DamageDict.GroupBy(entry => entry.Value, entry => entry.Key).MaxBy(entry => entry.Key)?.ToList() ?? []`
This 1) groups each mapping of damage type to amount into a bunch of bins, so anything with equal quantities gets put in the same key, 2) grabs the bin with the largest damage, 3) converts that to a list (which implicitly just includes the damage types), and then returns that or an empty list if MaxBy returned a null value.